hudi-common/src/main/java/org/apache/hudi/common/schema/HoodieFieldOrder.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.common.schema;
+
+import org.apache.avro.Schema;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Enumeration of field sort orders supported by Hudi schema system.
+ *
+ * <p>This enum wraps Avro's Schema.Field.Order to provide a consistent interface
+ * for field ordering while maintaining binary compatibility with Avro.</p>
+ *
+ * @since 1.2.0
+ */
+public enum HoodieFieldOrder {
```

Comment:
We use this in couple of places. One of them is AvroSchemaComparatorForSchemaEvolution where we check for equality for order.

Reply:
Is that really necessary though? I don't think we expose any option to set this to another value?

What is significance of field order?

Reply:
@the-other-tim-brown As we are using composition pattern (and not inheritance), the order would need to be supported. right ? Anyways, when you take over this PR, can you look into if this is needed and remove it if not needed.

Reply:
Will it make sense to have this as a per field value or does it make sense as a table property though?